File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascalSeries;
        pascalSeries.push_back({1});
        
        for(int row = 2; row <= numRows; row++) {
            vector<int> prevSeries = pascalSeries[pascalSeries.size() -1];
            vector<int> curSeries(prevSeries.size() +1, 1);
            
            for(int i = 1; i < prevSeries.size(); i++) {
                curSeries[i] = prevSeries[i -1] +prevSeries[i];
            }
            pascalSeries.push_back(curSeries);
        }
        
        return pascalSeries;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascalTriangle = new ArrayList<>();
        
        for(int i = 1; i <= numRows; i++) {
            List<Integer> prev = (i -2 >= 0)? pascalTriangle.get(i -2) : null;
            List<Integer> curr = new ArrayList<>();
            curr.add(1);
            
            for(int j = 1; j <= i -2; j++) {
                curr.add(prev.get(j -1) +prev.get(j));
            }
            
            if(i > 1) curr.add(1);
            pascalTriangle.add(curr);
        }
        
        return pascalTriangle;
    }
}